src/main/scala/uk/gov/nationalarchives/dp/client/DataProcessor.scala

```diff
@@ -134,32 +135,77 @@ class DataProcessor[F[_]]()(implicit me: MonadError[F, Throwable]) {
       case generationUrls => me.pure(generationUrls)
     }
 
+  /** Returns whether the the text content of every `Generations` -> `Generation` element
+    * @param generationsElement
+    *   The 'Generations' element containing the generations
+    * @return
+    *   A `Seq` of `String` with the text content of every `Generations` -> `Generation` element
+    */
+
+  def generationType(generationsElement: Elem, contentObjectRef: UUID): F[GenerationType] =
+    (generationsElement \ "Generation").map(_.attributes) match {
+      case Nil | List(xml.Null) =>
+        me.raiseError(PreservicaClientException(s"No attributes found for entity ref: $contentObjectRef"))
+      case attributesPerGenerationInList =>
+        val attributesPerGeneration = attributesPerGenerationInList.head
```

Comment:
I know the head is safe here because of the check above but just in case something changes, could we refactor this slightly.
If you do 
```scala
case (attributesPerGeneration: MetaData) :: _ =>
        val potentialOriginalityStatus = attributesPerGeneration.get("original").map(_.text)
```
It gets rid of the head line. What do you think?

Reply:
Ah, cool; looks neater. 👍